095-stretch-cluster.md

```diff
@@ -0,0 +1,328 @@
+
+Kubernetes [CoreDNS must also be reconfigured](https://aswinayyolath.github.io/stretch-kafka-docs/Setting-up-cilium/#setting-up-coredns-for-multi-cluster-dns-resolution) to allow DNS resolution across multiple Kubernetes clusters.
+
+#### Step 2: Deploy a cluster operator to each Kubernetes cluster
+
+##### Central cluster operator configuration
+
+The operator running in the central Kubernetes cluster must be provided with the following information to allow creation of resources in remote clusters:
+
+- The identifier for each Kubernetes cluster defined in 'Step 1'.
+- A URL endpoint for the Kubernetes API server running in each remote cluster.
+- Credential(s) to allow authentication with the remote Kubernetes API server(s).
+
+The information outlined above will be provided as an environment variable for the cluster operator.
+The value of the environment variable uses a "map" format as shown below:
+
+```yaml
+- name: STRIMZI_REMOTE_KUBE_CONFIG
+  value: |
+      cluster-id-a.url=<cluster-a URL>
+      cluster-id-a.secret=<secret-name-cluster-a>
+      cluster-id-b.url=<cluster-b URL>
+      cluster-id-b.secret=<secret-name-cluster-b>
+```
+
+The secrets referenced here must contain the kubeconfig for the Kubernetes cluster available at the provided URL as the value of secret key 'kubeconfig'.
+This allows the central Strimzi operator to authenticate with multiple Kubernetes clusters.
+
+**Example Secret**
+
+Below is an example Kubernetes Secret containing a kubeconfig for a remote cluster
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-name-cluster-a
+  namespace: strimzi
+type: Opaque
+data:
+  kubeconfig: <base64-encoded-kubeconfig>
+```
+
+When this environment variable is set, the cluster operator will understand it needs to deploy a stretch Kafka cluster.
+The custom actions taken as a result of this environment variable include:
+1. The value of `STRIMZI_NETWORK_POLICY_GENERATION` is assumed to be `false` irrespective of the value set by the user.
+This is to ensure network traffic between Kubernetes clusters is not blocked.
+2. The `StrimziPodSet` resources created in a remote cluster will include an annotation `strimzi.io/orphaned-podset: true`.
+This annotation will allow the remote cluster operator to reconcile the `StrimziPodSet` without a `Kafka` and `KafkaNodePool` CR being present in the same cluster.
+
+##### Remote cluster operator configuration
+
+The operator running in the remote cluster must be set to only reconcile `StrimziPodSet` resources by setting the existing environment variable:
+
+```yaml
+- name: STRIMZI_POD_SET_RECONCILIATION_ONLY
+  value: true
+```
+
+This configuration will minimise the resources required by the cluster operator on the remote Kubernetes clusters and simplify logging output.
+
+#### Step 3: Create Kafka and KafkaNodePool resources in the central cluster
+
+##### Kafka CR
+
+The `Kafka` CR must include a new annotation to specify the chosen Cloud native network project:
```

Comment:
can you add an explanation about why this annotation is needed. My guess is that the operator has to deal with some network project specific configuration/interaction but it would be better to give more info here.
Also, what happens if the users forgot to set it? Still one more thing to clarify.